spec/lib/problem_check_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+RSpec.describe ProblemCheck do
+  # rubocop:disable RSpec/BeforeAfterAll
+  before(:all) do
```

Comment:
Just wondering why bother with this before + after all, and not just define the class inside the spec?

Reply:
Since each subclass created becomes a problem check, if it's not undefined, it will leak into other tests and give order-dependent flakiness.

Reply:
Ah okay cool, makes sense :+1: